nucypher/network/server.py

```diff
@@ -241,23 +240,18 @@ def set_policy(id_as_hex):
             return Response("Invalid Signature", status_code=400)
 
         if not this_node.federated_only:
-            # This splitter probably belongs somewhere canonical.
-            transaction_splitter = BytestringSplitter(32)
-            tx, kfrag_bytes = transaction_splitter(cleartext, return_remainder=True)
+            # TODO: This splitter probably belongs somewhere canonical.
+            from nucypher.policy.policies import Policy
+            policy_id_splitter = BytestringSplitter(Policy.POLICY_ID_LENGTH)
+            hrac, kfrag_bytes = policy_id_splitter(cleartext, return_remainder=True)
 
             try:
-                # Get all of the arrangements and verify that we'll be paid.
-                # TODO: We'd love for this part to be impossible to reduce the risk of collusion.  #1274
-                arranged_addresses = this_node.policy_agent.fetch_arrangement_addresses_from_policy_txid(tx, timeout=this_node.synchronous_query_timeout)
-            except TimeExhausted:
+                this_node.verify_policy_payment(policy_id=hrac, timeout=this_node.synchronous_query_timeout)
```

Comment:
Is hrac in `bytes` or a `str`; presumably `bytes`? The typehint for `policy_id` is `str`.